src/python/pants/backend/graph_info/subsystems/cloc_binary.py

```diff
@@ -12,3 +15,13 @@ class ClocBinary(Script):
 
   replaces_scope = 'cloc'
   replaces_name = 'version'
+
+  @memoized_classproperty
+  def default_versions_and_digests(cls):
```

Comment:
It feels like the various definitions could be true class properties, rather than being lazily computed? ie:
```
class ClocBinary(..):
  ..
  default_version = '1.80'
  default_versions_and_digests = {
      ..,
    }
```

Is there any particular reason you went with method+decorator?

Reply:
To DRY the use of `default_version` by pulling it from `cls`! I think it makes more sense to have that repetition now, though, so I'll write it inline as you suggest.